Extension/src/LanguageServer/configurations.ts

```diff
@@ -1001,7 +1001,9 @@ export class CppProperties {
                 }
             }
 
-            configuration.customConfigurationVariables = this.updateConfigurationStringDictionary(configuration.customConfigurationVariables, settings.defaultCustomConfigurationVariables, env);
+            if (settings.defaultCustomConfigurationVariables !== null) {
```

Comment:
If setting.defaultCustomConfigurationVariables is originally non-null and then set to null, this check appears to not set it to null (or undefined)? That seems incorrect, but I haven't tried it out.